crypto/store/loader_file.c

```diff
@@ -272,6 +272,7 @@ static OSSL_STORE_INFO *try_decode_PKCS12(const char *pem_name,
                         osi_ca = NULL;
                         (void)sk_X509_shift(chain);
                     }
+                    sk_X509_free(chain);
```

Comment:
Shouldn't this "free" happen after the following "}" not before?

Reply:
Right, the fix was not sufficient.
Having an even closer look I found a couple of more potential memory leaks in case of allocation errors. Moreover the function did not cope with cases that the pkey and/or cert in the PKCS#12 structure is not present.
I've fixed all these issues.